src/riscv/lib/src/state_backend/proof_backend/proof/deserialiser.rs

```diff
@@ -185,6 +188,26 @@ pub trait Suspended {
     where
         Self::Output: 'static;
 }
+
+/// Trait for common deserialisation functionality abstracting over the deserialiser type.
+pub trait RunDeserialiser: Deserialiser + Sized {
+    /// Data type consumed by the deserialiser.
+    type Data;
+
+    /// Run the deserialiser and obtain the result of the computation.
+    fn run<R>(
+        input_data: Self::Data,
+        deser_fn: impl FnOnce(Self) -> DeserResult<<Self as Deserialiser>::Suspended<R>>,
+    ) -> Result<R>;
+
+    /// Based on a [`ProofLayout`] obtain the corresponding [`AllocatedOf`] and [`OwnedProofPart`].
+    fn into_verifier_alloc<L: ProofLayout>(
+        input_data: Self::Data,
+    ) -> Result<(AllocatedOf<L, Verifier>, OwnedProofPart)> {
+        Self::run(input_data, L::into_verifier_alloc)
+    }
+}
```

Comment:
This trait is only used in a monomorphic way, i.e. in situations where you know the trait implementor. Is the trait actually needed?

Reply:
True, I could change it, but the trait allows me to generically write a `into_verifier_alloc` function for all deserialisers. Otherwise I'd use a generic `run` method and have to cast the layout manually to a `ProofLayout`. 
For comparison, thougts?:

functions:
```
deserialise_owned::run(
    ProofTree::Present(proof),
    <PvmLayout<MC, BCC> as ProofLayout>::into_verifier_alloc,
)
// or
deserialise_stream::run(
    tree_serialisation.as_ref(),
    <PvmLayout<MC, BCC> as ProofLayout>::into_verifier_alloc
)
```

trait:
```
ProofTreeDeserialiser::into_verifier_alloc::<PvmLayout<MC, BCC>>(proof_tree)
// or
StreamDeserialiser::into_verifier_alloc::<PvmLayout<MC, BCC>>(&tree_serialisation)
```